src/ui/actions/app.ts

```diff
@@ -118,38 +120,59 @@ export function setupApp(recordingId: RecordingId, store: UIStore) {
   ThreadFront.ensureProcessed("basic", undefined, regions =>
     store.dispatch(onUnprocessedRegions(regions))
   ).then(() => {
-    store.dispatch({ type: "loading", loading: 100 });
+    store.dispatch(setLoading(100));
   });
 
   ThreadFront.ensureProcessed("executionIndexed").then(() => {
     console.log("indexed");
-    store.dispatch({ type: "indexed" });
+    store.dispatch(setIndexing(100));
   });
 
   ThreadFront.listenForLoadChanges((parameters: loadedRegions) =>
     store.dispatch({ type: "set_loaded_regions", parameters })
   );
 }
 
-function onUnprocessedRegions({ regions }: unprocessedRegions): UIThunkAction {
+function onUnprocessedRegions({ level, regions }: unprocessedRegions): UIThunkAction {
   return ({ dispatch, getState }) => {
-    const loading = selectors.getLoading(getState());
-    const endPoint = Math.max(...regions.map(r => r.end), 0);
+    let endPoint = Math.max(...regions.map(r => r.end), 0);
     if (endPoint == 0) {
       return;
     }
+    const state = getState();
+    if (endPoint > selectors.getRecordingDuration(state)) {
```

Comment:
Another option would be to store the loaded regions in redux which we'll want to use in the future to show what is accessible in the timeline and console.

I'll go ahead and land this PR and then we can add regions as a follow up